test/shared/src/main/scala/zio/test/TestAspect.scala

```diff
@@ -168,6 +168,35 @@ object TestAspect extends TimeoutVariants {
       }
     }
 
+  /**
+   * An aspect that makes a test succeed if it fails for any cause.
+   */
+  val failure: PerTest[Nothing, Any, Nothing, Any, Unit, Unit] = failure(Assertion.anything)
+
+  /**
+   * An aspect that makes a test succeed if it fails for a given cause.
+   */
+  def failure[E0](p: Assertion[TestFailure[E0]]): PerTest[Nothing, Any, Nothing, E0, Unit, Unit] =
+    new TestAspect.PerTest[Nothing, Any, Nothing, E0, Unit, Unit] {
+      def perTest[R >: Nothing <: Any, E >: Nothing <: E0, S >: Unit <: Unit](
+        test: ZIO[R, TestFailure[E], TestSuccess[S]]
+      ): ZIO[R, TestFailure[E], TestSuccess[S]] = {
+        def fail(value: TestFailure[E]) = ZIO.fail(
```

Comment:
Maybe you can inline these definitions into `test.foldM` since they are each only used once?